Basic idea prototype

marc-jones opened the pull request:
The black hole exerts a force on the ship proportional to their masses and the inverse square distance between them.

The black hole can be moved by clicking on the play area.

The ship can move infinitely to the right of the screen, but is limited vertically and cannot backtrack.

tomcraven committed:
initial commit of prototyping

tomcraven committed:
ship + black hole, ready to physics dat shit up

Commit:
Made the ship orbit a point and slowly fall out of that orbit

marc-jones committed:
Got the ship to orbit around the black hole

Turns out that you can set local gravity on an object and that
sorts out all the complex maths required to work out how the
ship's angular velocity and linear velocity changes

marc-jones committed:
Clicking now moves the black hole

marc-jones committed:
Added a camera which follows the ship

marc-jones committed:
Made the world infinite to the right of the camera

When the ship moves beyond a certain x direction, its x position
ceases to change and the other objects positions' change to give
the illusion that the ship is moving to the right. An array stores
all non player objects and updates them all if the ship reaches the
set certain threshold.

I've also done some cleaning up of the formatting to restrict all
lines to a maximum of 79 characters.

tomcraven commented on src/states/game.js:
```diff
@@ -0,0 +1,102 @@
+                                           new Phaser.Point(0, 0)));
+        
+        if (this.input.activePointer.leftButton.isDown) {
+            blackHole.position = new Phaser.Point(
```
Would be good to avoid creating a new point here, can we use blackHole.position.setTo instead?

marc-jones replied:
Done

tomcraven commented on src/states/game.js:
```diff
@@ -0,0 +1,102 @@
+
+        ship.body.rotation = 270 + Phaser.Math.radToDeg(
+            Phaser.Math.angleBetweenPoints(ship.body.velocity,
+                                           new Phaser.Point(0, 0)));
```
Instead of new-ing this one each update call we should create a constant in 'create' and use it here.

this.__origin, or this.__zeroPoint (or whatever) = new Phaser.Point(0,0);

I had a google and i'm surprised there's not a Phaser.Point.ZERO or something similar

marc-jones replied:
I took your advice and created a this.__origin variable. You're right that it's a bit weird, but I think your solution will work well.

tomcraven commented on src/states/preload.js:
```diff
@@ -0,0 +1,28 @@
+	}
+};
+
+Core.addState( STATE_NAME.PRELOAD, Preload )
```
github is whinging about the lack of new line at the end of the file, really don't think it should be an issue though

marc-jones replied:
I've added newline characters to the files GitHub was complaining about.

tomcraven commented on src/states/game.js:
```diff
@@ -0,0 +1,102 @@
+        var angleBetweenShipBH = Phaser.Math.angleBetweenPoints(ship.position,
+            blackHole.position)
+        
+        ship.body.gravity = new Phaser.Point(totalGravity *
```
should be able to do gravity.setTo here as well

marc-jones replied:
Done

marc-jones committed:
Some optimization in the update function and code cleaning

Changed the code to remove the creation of new Phaser points each
update step. Also added newline characters to the end of each
source file.

tomcraven commented:
Looks good!
